Plutarch/List.hs

```diff
@@ -261,12 +260,12 @@ pzip ::
   , PElemConstraint list (PPair a b)
   ) =>
   Term s (list a :--> list b :--> list (PPair a b))
-pzip = pzipWith' $ \x y -> pcon (PPair x y)
+pzip = phoistAcyclic $ pzipWith' $ \x y -> pcon (PPair x y)
```

Comment:
curious question: would using `pzipWith #$ plam $ \x y -> pcon (PPair x y)`  be more efficient here or would it not matter?

Reply:
`pzipWith'` is more efficient, since we only use the terms once. In general I feel like HOFs should be Haskell-level.